src/idl_gen_cpp.cpp

```diff
@@ -179,6 +171,18 @@ class CppGenerator : public BaseGenerator {
         code += name + "Identifier()); }\n\n";
       }
 
+      // The root verifier:
+      code += "inline bool Verify";
+      code += name;
+      code +=
+          "Buffer(flatbuffers::Verifier &verifier) { "
+          "return ";
+      if (parser_.file_identifier_.length())
+          code += "verifier.GetInternalBufferSize() > (2 * sizeof(flatbuffers::uoffset_t)) && " + name + "BufferHasIdentifier(verifier.GetInternalBuffer()) && ";
```

Comment:
Move this code into Verifier::VerifyBuffer. You can pass the file_identifier (or nullptr) as an argument.